### PR TITLE
Add induction command

### PR DESCRIPTION
**Pull request by sapphyree**
As per @rainestormee's suggestion, adds an `/induct` command to automatically give a user the Guest role whilst sending a message in general to highlight the fact they've joined and to draw their attention to the new channels.

### CODE REVIEW

**Review by MattyTheHacker — APPROVED** [review]
Personally I think it's better to have this done by automatically checking reacts but this works too

**Comment by sapphyree** [comment]
> Personally I think it's better to have this done by automatically checking reacts but this works too

The problem with automatically checking reacts is that you're spending processing time constantly checking all of the messages in introductions. In addition, you're checking every user that reacts (because there's no way to just filter) and that takes up a lot of time checking the cache (or, in some cases, performaing an async fetch and re-checking the cache). This way is more annoying, yes, but it's easier to not have to worry about checking every message in the channel.

**Comment by MattyTheHacker** [comment]
Very good points, can't argue with that

File: src/commands/makeMember.js [hunk 1]
```diff
@@ -40,13 +41,15 @@ module.exports = {
...
+      console.log(
+        `${dt} - Warning: ${interaction.user.tag} is already a member`
+      );
       return await interaction.editReply({
         content: "You're already a member - why are you trying this again?",
       });
     } else if (this.usedIDs.includes(enc)) {
       console.log(
-        `Warning: ${interaction.user.tag} tried using an ID that has already been used.`
+        `Warning: $${dt} - {interaction.user.tag} tried using an ID that has already been used.`
```
Comment (LMBishop) [inline src/commands/makeMember.js:52]:
```suggestion
        `Warning: ${dt} - ${interaction.user.tag} tried using an ID that has already been used.`
```
Typo

File: src/events/interactionCreate.js [hunk 2]
```diff
@@ -14,8 +10,8 @@ module.exports = {
     try {
       command.execute(interaction);
     } catch (err) {
-      console.error(err);
-      interaction.reply({
+      console.error(`${dt} - ${err}`);
+      interaction.replsy({
         content: "There was an errour while executing this command.",
```
Comment (LMBishop) [inline src/events/interactionCreate.js:15]:
```suggestion
      interaction.reply({
        content: "There was an error while executing this command.",
```
Typo

File: src/commands/induct.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    } catch (error) {
+      console.log(`${dt} - Error: #roles does not exist on this guild.`);
+    }
+
+    let message = `Hey everyone! Please welcome ${user} to CSS - remember to grab your roles in ${roles} and say hello to everyone here!`;
+
+    try {
+      role = interaction.guild.roles.cache.find((r) => r.name === "Guest");
+    } catch (error) {
+      console.log(`{dt} - Error: @Guest does not exist on this guild.`);
```
Comment (LMBishop) [inline src/commands/induct.js:52]:
```suggestion
      console.log(`${dt} - Error: @Guest does not exist on this guild.`);
```
Typo

**Review by LMBishop — COMMENTED** [review]
The only concern I have is that the welcome message could get spammy, especially during welcome week.

Also, not really that important, but something to consider is a logging library for less 'fragile?' (for lack of a better term) logs.

**Comment by LMBishop** [comment]
Oooo I've just had an idea, I know you've already implemented it like this, *buuut* I've seen an 'Apps' thing when you right click messages on Discord and I think it would be pretty cool to have one of those also trigger this

image

**Comment by MattyTheHacker** [comment]
> Oooo I've just had an idea, I know you've already implemented it like this, _buuut_ I've seen an 'Apps' thing when you right click messages on Discord and I think it would be pretty cool to have one of those also trigger this

Seconded, that would be pretty cool

**Comment by MattyTheHacker** [comment]
> The only concern I have is that the welcome message could get spammy, especially during welcome week.
>

Could potentially add an option to the command where the user can specify whether they want it to send a message in general or not

**Comment by sapphyree** [comment]
> > The only concern I have is that the welcome message could get spammy, especially during welcome week.
> >
> 
> Could potentially add an option to the command where the user can specify whether they want it to send a message in general or not? Partially because I think it should DM them as well in all cases also to remind them to change their nickname and to grab roles etc...
> 
> https://github.com/CSSUoB/TeX-Bot/issues/37
> 
> #8 

The DMing functionality is OoS for this PR and this command right now, though I was thinking of adding a `silent` Boolean option yes.

**Comment by JTWWilson** [comment]
Re: @LMBishop's concerns about spamming, I think the best approach would be to enable the functionality and see how annoying it is. Anything else is conjecture and I don't think that enabling it for a short amount of time is much of an issue.

**Comment by sapphyree** [comment]
Re: @LMBishop's suggestion with the logging library - I agree, in the future I'd like to move to a better solution to logging events, but for right now I think it's probably fine? Feel free to open an issue and/or implement it yourself <3

**Pull request merged**